example/bar3d_example.py

```diff
@@ -27,4 +27,102 @@ def bar3d_base() -> Bar3D:
     return c
 
 
+@C.funcs
+def bar3d_stack() -> Bar3D:
+    def generate_data():
```

Comment:
A personal choice here. I would avoid nested function because it is hard to test.